mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs

```diff
@@ -232,20 +232,27 @@ void ConstructCRef (StringBuilder sb)
 			if (DescKind == Kind.Namespace)
 				return;
 
-			sb.Append ('.');
+			if (!string.IsNullOrEmpty (Namespace))
+				sb.Append ('.');
+
 			sb.Append (TypeName);
 			if (GenericTypeArguments != null) {
-				sb.Append ('<');
-				int i=0;
-				foreach (var t in GenericTypeArguments) {
-					if (i > 0) {
-						sb.Append (",");
+				if (GenericTypeArguments.Any (a => a == null)) {
```

Comment:
I don't like this, because it's not particularly clear why "if any argument is null, use the count".

It also seems duplicative; there's *already* a `EcmaDesc.GenericTypeArgumentsIsNumeric` property, which seems to have the same *intent* behind it; shouldn't `GenericTypeArgumentsIsNumeric` be used here, instead of `.Any()`?

Also, `GenericTypeArgumentsIsNumeric` should probably be updated to use `.Any()` instead of `.FirstOrDefault()`.